packages/adblocker/src/engine/engine.ts

```diff
@@ -1498,13 +1525,80 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
         if (result.filter !== undefined) {
           result.exception = this.exceptions.match(request, this.isFilterExcluded.bind(this));
         }
+
+        // If the request is allowed in any way, we match request substitudes.
+        if (result.filter === undefined || result.exception !== undefined) {
+          const substitutions: Map<NetworkFilter, NetworkFilter | undefined> = new Map();
+          const url = new URL(request.url);
```

Comment:
Nit. Let's only parse the URL if we have a match since we won't have any for the vast majority of calls.

Reply:
Updated in 22cd3d71075ce5279916e6b88f56fbbde657beee but not sure if it matches your thought. Please, check the following section: [`22cd3d7` (#4528)](https://github.com/ghostery/adblocker/pull/4528/commits/22cd3d71075ce5279916e6b88f56fbbde657beee#diff-0a5765da7fa865c0ab71fc7b0187e3c6663bcc602558022d04dd7da3c0ab410aR1540-R1549)